cpp/include/cudf/copying.hpp

```diff
@@ -173,5 +174,31 @@ std::vector<column_view> slice(column_view const& input,
 std::vector<column_view> split(column_view const& input,
                                std::vector<size_type> const& splits);
 
+/**
+* @brief Creates a new `table` similar to the input but with values in every
+* column offset by `periods`, using `fill_value` when input is out-of-bounds.
+*
+* When `i + periods` is in bounds:
+*   `out[col][i] = in[col][i + periods]`
```

Comment:
So this is the reverse of pandas? If out[i] == in[i + period] then for in = [1,2,3,4,5], period = 2, out = [3,4,5,None,None]? But in Pandas, the shift goes in the other way

Reply:
Addressing these concerns in #3662.